PetAdoption-iOS/PetAdoption-iOS/Controllers/filterViewControlelr.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  filterViewControlelr.swift
+//  PetAdoption-iOS
+//
+//  Created by Amir Fleminger on 4/11/18.
+//  Copyright © 2018 Code For Orlando. All rights reserved.
+//
+
+import UIKit
+
+class FilterViewController: UITableViewController {
+    var delegate: FilterSelectorDelegate?
+    enum RowIndex:Int {
+        case All = 0, Dogs, Cats, Birds, SmallFurry, Horses, BarnYard, Reptiles, Rabbits
+    }
+    
+    var selectedAnimalTypes: Set<String> = []
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Adding a footer removes redundant cell dividers from the tableview
+        tableView.tableFooterView = UIView(frame: .zero)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        selectedAnimalTypes = FilterViewController.loadAnimalTypesSelected()
+        setAnimalTypeCheckmarks()
+    }
+    
+    class func loadAnimalTypesSelected() -> Set<String>{
+        let defaults = UserDefaults.standard
+        
+        guard let data:Data = defaults.object(forKey: "selectedAnimalTypes") as? Data else {
+            print("Couldnt load animal type preferences! Default to all animal types")
+            return ["all"]
+        }
+        // convert the Data object to a Set<String>
+        return NSKeyedUnarchiver.unarchiveObject(with: data) as! Set<String>
+    }
+    
+    class func saveAnimalTypesSelected(_ selectedAnimalTypes_: Set<String>){
+        let defaults = UserDefaults.standard
+        let encodedData = NSKeyedArchiver.archivedData(withRootObject: selectedAnimalTypes_)
+        defaults.set(encodedData, forKey: "selectedAnimalTypes")
+        defaults.synchronize()
+    }
+    
+    func setAnimalTypeCheckmarks() {
```

Comment:
Another bug I've noticed is that you can get into a situation where nothing is checked at all.  If I select an animal type and then deselect it, now nothing is checked.  It would be nice if in the case that nothing is checked, the "all" option becomes checked.  Is that possible?

Reply:
Fixed with 50a3b0d